pkg/volume/hostpath/host_path.go

```diff
@@ -305,6 +305,21 @@ func (r *hostPathProvisioner) Provision(selectedNode *v1.Node, allowedTopologies
 			},
 		},
 	}
+
+	var labels map[string]string
+	if r.options.PVC.Spec.Selector != nil {
+		labels = r.options.PVC.Spec.Selector.MatchLabels
+	}
```

Comment:
We do not check pvc label, but we should check pvc Selector

Reply:
Selector can contain different things than `MatchLabels`.

Every other provisioner does not support `PVC.Spec.Selector`. The reason is that labels have meanings. If I have a PVC with selector `department not in [accounting, marketing]`, the provisioner must understand what the label means, what departments are there and choose e.g. `engineering`. How can the provisioner know that?

https://github.com/kubernetes/kubernetes/blob/5a3301a59d0c4d5e76002cff99a2f5450d2eea80/pkg/volume/gcepd/gce_util.go#L139-L141

What is the use case for your change? What selectors should the HostPath provisioner understand? HostPath provisioner was created for debugging, I hope nobody uses it in production (hint: use local storage).

Reply:
Thanks,  I just run hostpath on my single-node test cluster.  I just test it.

Would we support this in the future?